src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java

```diff
@@ -0,0 +1,47 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import java.util.function.Supplier;
+
+/**
+ * driver adapter to drawer with several bugs.
+ */
+public class LineDrawerAdapter extends DrawPanelController implements Job2dDriver {
+	private int startX = 0, startY = 0;
+	private DrawPanelController drawerController;
+	private Supplier<ILine> lineSupplier;
+
+	public LineDrawerAdapter(DrawPanelController drawerController) {
+		super();
+		this.drawerController = drawerController;
+		setLineSupplier(LineFactory::getBasicLine);
```

Comment:
Direct coupling with LineFactory class is unnecessary, as well as using a setter from within the current class.